elasticsearch_dsl/field.py

```diff
@@ -114,17 +118,30 @@ class Object(Field):
     name = 'object'
     _coerce = True
 
-    def __init__(self, doc_class=None, **kwargs):
-        self._doc_class = doc_class
+    def __init__(self, doc_class=None, dynamic=None, properties=None, **kwargs):
+        """
+        :param document.InnerDoc doc_class: base doc class that handles mapping.
+            If no `doc_class` is provided, new instance of `InnerDoc` will be created,
+            populated with `properties` and used
```

Comment:
I feel like, when we are defining args we should also check them - providing `properties` and `doc_class` together should raise an exception as it is not a valid combination.

Previously we could have pretended that we don't know as we are just passing things through, but now we no longer have the excuse :)